purge-cluster.yml

```diff
@@ -2,42 +2,234 @@
+      ansible_distribution == 'Ubuntu' and
+      osd_group_name in group_names
 
-  - name: disk zap again
-    command: /usr/sbin/sgdisk --zap-all --clear --mbrtogpt -g -- {{ item }}
-    with_items: devices
+  - name: stop ceph mons on ubuntu
+    command: stop ceph-mon-all
+    failed_when: false
+    when:
+      ansible_distribution == 'Ubuntu' and
+      mon_group_name in group_names
+  
+  - name: stop ceph mdss on ubuntu
+    command: stop ceph-mds-all
+    failed_when: false
+    when:
+      ansible_distribution == 'Ubuntu' and
+      mds_group_name in group_names
+
+# rc is 2 if file not found, so no mount point, so no error
+  - name: get osd data mount points
+    shell: ls /var/lib/ceph/osd
+    register: mounted_osd
+    changed_when: false
+    failed_when: mounted_osd.rc != 0 and mounted_osd.rc != 2
+    when:
+     osd_group_name in group_names
+
+  - name: umount osd data partition
+    shell: umount /var/lib/ceph/osd/{{ item }}
     failed_when: false
+    with_items:
+     - "{{ mounted_osd.stdout_lines }}"
+    when:
+     osd_group_name in group_names
+
+  - name: zap osd disks
+    shell: ceph-disk zap "{{ item }}"
+    with_items: devices
+    when:
+     osd_group_name in group_names
+
+  - name: purge ceph packages with yum
+    yum:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_packages }}"
+    when:
+      ansible_pkg_mgr == 'yum'
+
+  - name: purge ceph packages with dnf
+    dnf:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_packages }}"
+    when:
+      ansible_pkg_mgr == 'dnf'
+
+  - name: purge ceph packages with apt
+    apt:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_packages }}"
+    when:
+      ansible_pkg_mgr == 'apt'
+
+  - name: purge remaining ceph packages with yum
+    yum:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_remaining_packages }}"
+    when:
+      ansible_pkg_mgr == 'yum' and
+      purge_all_packages == true
+
+  - name: purge remaining ceph packages with dnf
+    dnf:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_remaining_packages }}"
+    when:
+      ansible_pkg_mgr == 'dnf' and
+      purge_all_packages == true
+
+  - name: purge remaining ceph packages with apt
+    apt:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_remaining_packages }}"
+    when:
+      ansible_pkg_mgr == 'apt' and
+      purge_all_packages == true
+
+  - name: remove config
+    file:
+     path: /etc/ceph
+     state: absent
+
+  - name: remove data
+    file:
+     path: /var/lib/ceph
+     state: absent
+
+  - name: remove logs
+    file:
+     path: /var/log/ceph
+     state: absent
+
+  - name: remove form SysV
+    shell: "update-rc.d -f ceph remove"
+    when:
+      ansible_distribution == 'Ubuntu'
+
+  - name: remove Upstart nad SysV files
+    shell: 'find /etc -name "*ceph*" -delete'
```

Comment:
Why do you sometimes use simple or double quotes for the shell module?
Double quotes all the time are probably the way to go.